spec/unit/action/deploy/environment_spec.rb

```diff
@@ -273,16 +282,18 @@
         let(:purge_levels) { [:environment] }
 
         it "only logs about purging environment" do
+          expect(subject).to receive(:visit_environment).and_wrap_original do |original, env, &block|
+            expect(env.logger).to_not receive(:debug).with(/purging unmanaged puppetfile content/i)
+            original.call(env)
+          end.at_least(:once)
+
           expect(subject.logger).to receive(:debug).with(/purging unmanaged content for environment/i)
           expect(subject.logger).to_not receive(:debug).with(/purging unmanaged environments for deployment/i)
-          expect(subject.logger).to_not receive(:debug).with(/purging unmanaged puppetfile content/i)
 
           subject.call
         end
 
         it "logs that environment was not purged if deploy failed" do
-          expect(subject).to receive(:visit_puppetfile) { subject.instance_variable_set(:@visit_ok, false) }
```

Comment:
While I'm not sure what is wrong with the first test of these two. This test, I think, is failing because it depends on the instance variable `@visit_ok` being set to false during the deploy. However, I attempted this with no success:
```rb
expect(subject).to receive(:visit_environment).and_wrap_original do |original, env, &block|
  expect(env).to receive(:deploy) { subject.instance_variable_set(:@visit_ok, false) }
  original.call(env)
end
```
without success. (Not sure why I switched the test content back)

I'm wondering if I'm missing something simple in the first test case and if figured out then this one would be fixable by the above code block (or similar)

Reply:
Is this out of date now? All the tests seem to be passing at this point.

Reply:
Yeah, this is out of date now.